### DISCUSSION TITLE
Why does the web binding check if it's running in Node?

### DISCUSSION THREAD

**Discussion by gushogg-blake** [tree-sitter/tree-sitter#3970] [Q&A]
In my Electron-based editor I changed the code around [this line](https://github.com/tree-sitter/tree-sitter/blob/cd6e7663653e8de6624a40cb624b3419b07a06c3/lib/binding_web/binding.js#L1283) to be the other way around -- first check if we're in a browser then assume we're in Node.

This is because the Node check returns true in the browser for Electron and it was breaking the language loading. I was wondering what the intention was here - is this logic specifically for Electron, or something else?

**Update** use commit to avoid line reference going out of date

**Comment by amaanq** [comment]
I believe it's for node.js-based applications, a PR to fix this for electron would be welcome
